lab_1_keywords_tfidf/main.py

```diff
@@ -99,7 +151,25 @@ def calculate_tfidf(term_freq: dict[str, float], idf: dict[str, float]) -> Optio
 
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if not isinstance(term_freq, dict) or not isinstance(idf, dict):
+        return None
+    if term_freq == {}:
+        return None
+
+    for k, val in term_freq.items():
+        if not isinstance(k, str) or not isinstance(val, float):
+            return None
+        if idf == {}:
```

Comment:
bad condition. an empty dict equals `False` - this is a Falsy object

Reply:
fixed